update LLVM docs (#56162)
JuliaLang/julia: doc/src/devdocs/llvm.md

fun, T = +, Tuple{Int,Int} # Substitute your function of interest here
optimize = false
open("plus.ll", "w") do file

>     println(file, InteractiveUtils._dump_function(fun, T, false, false, true, true, :att, optimize, :default, false))

Reviewer: I am unsure why we don't just do

``code_llvm(io, fun, T, raw=true, dump_module=true)` instead of using this lower level function.

Author: probably because `code_llvm` dumps to your terminal rather than to a string so redirecting it to a file is harder.

Reviewer 2: What Valentin is saying is that you can do
```julia
open("/tmp/exp.ll", "w") do io
    code_llvm(io, exp, (Float64,); raw=true, dump_module=true)
end
```
because `code_llvm` does take an `IO` as input

Reviewer 2: Concretely, his suggestion was
```suggestion
    code_llvm(file, fun, T; raw=true, dump_module=true, optimize)
```

end
```
These files can be processed the same way as the unoptimized sysimg IR shown
above, or if you want to see the LLVM IR yourself and get extra verification run, you can use
```

> ./usr/tools/opt -load-pass-plugin=libjulia-codegen.so --passes='julia' -S -verfiy-each plus.ll

Reviewer: ```suggestion
./usr/tools/opt -load-pass-plugin=libjulia-codegen.so --passes='julia' -S -verifiy-each plus.ll
```

> ./usr/tools/opt -load-pass-plugin=libjulia-codegen.so --passes='julia' -S -verifiy-each plus.ll

Reviewer 2: ```suggestion
./usr/tools/opt -load-pass-plugin=libjulia-codegen.so --passes='julia' -S -verify-each plus.ll
```

```

> (note on MacOS this would be `libjulia-codegen.dynlib` and on Windows, you should probably use WSL)

Reviewer 3: ```suggestion
(note on MacOS this would be `libjulia-codegen.dylib` and on Windows `libjulia-codegen.dll`)
```
